.github/workflows/main.yml

```
@@ -62,12 +53,12 @@ jobs:
       - name: Install python dependencies
         run: |
           pip install --user --upgrade pip
-          pip install tox
+          pip install pre-commit
           pip --version
-          tox --version
+          pre-commit --version
 
-      - name: Run tox
-        run: tox
+      - name: Run pre-commit hooks
+        run: pre-commit
```

Should this be something like:
```suggestion
        run: pre-commit --all-files --show-diff-on-failure
```
There are no staged files when this workflow runs.

---

Oooh good catch (`show-diff-on-failures` doesn't seem to be a thing however).  Fixed!

---

nice! and more on `--show-diff-on-failure` [here](https://pre-commit.com/#pre-commit-run). totally optional but I've found it helpful in CI

---

`failure` singular!  whomp whomp adding it now.